### PR TITLE
ci: deploy to preview all apps

### PR DESCRIPTION
**Pull request by lianghx-319**
## Why
To preview changes quickly when PR and deploy github pages for demo preview

## How
Describe your steps:
1. add a deploy script
2. build all template and deploy

**X-Mind**
image

## Test
**Travis-CI**
- Remember to add github pages URL as PUBLIC_PATH in CI Environment Variable

image

**Netlify**
- [deploy preview](https://deploy-preview-3--create-nuxt-app.netlify.com)
- netlify build settings like this
image

image

image

### CODE REVIEW

File: bin/deploy/build.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+const afterBuild = templateList.reduce((p, item) => {
+  const ps = new BuildProcess(item, { env: env[item] })
+  p = p.then(() => ps.run())
+    .then(() => templateData.push(item))
+  return p
+}, Promise.resolve())
+
+afterBuild
+  .then(() => buildIndex(templateData, deployDir('index.html')))
+  .catch((e) => {
```
Comment (2nthony) [inline bin/deploy/build.js]: 🤔xo(eslint) 规范catch的参数应命名为 `error`

File: bin/deploy/build.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env node
+const path = require('path')
+const glob = require('glob')
+const fse = require('fs-extra')
+const _ = require('lodash')
+const spawn = require('cross-spawn')
+const env = require('./env.json')
+
+const deployDir = (name = '') => path.join(__dirname, '../../dist', name)
```
Comment (levy9527) [inline bin/deploy/build.js:9]: 调整一下顺序，把函数定义放后面，实际执行逻辑代码放前面

Comment (2nthony) [reply bin/deploy/build.js:9]: 如果函数放底部建议命名为 `getDeployDir`

**Review by levy9527 — APPROVED** [review]

**Pull request merged**